src/platform/api/pythonApi.ts

```diff
@@ -345,28 +355,23 @@ export class InterpreterService implements IInterpreterService {
         this.hookupOnDidChangeInterpreterEvent();
         return this.didChangeInterpreters.event;
     }
+    private readonly _interpreters = new Map<string, { resolved: PythonEnvironment }>();
+    public get resolvedEnvironments(): PythonEnvironment[] {
+        this.hookupOnDidChangeInterpreterEvent();
+        return Array.from(this._interpreters.values()).map((item) => item.resolved);
+    }
+    public get environments(): readonly PythonEnvironmentV2[] {
+        this.getApi().catch(noop);
+        this.hookupOnDidChangeInterpreterEvent();
+        return this.api?.environments?.known || [];
+    }
 
-    @traceDecoratorVerbose('Get Interpreters', TraceOptions.Arguments | TraceOptions.BeforeCall)
-    public getInterpreters(): Promise<PythonEnvironment[]> {
-        const stopWatch = new StopWatch();
+    private getInterpreters(): Promise<PythonEnvironment[]> {
```

Comment:
The method `getInterrpeters` is no longer used, we will not rely on a push model completely.

Reply:
Fortunately this was used only in a couple of places in the extension code.